hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -277,34 +273,31 @@ collectComTx networkId scriptRegistry vk initialThreadOutput commits headId =
       headDatumAfter
       ReferenceScriptNone
   headDatumAfter =
-    mkTxOutDatum
+    mkTxOutDatumInline
       Head.Open
         { Head.parties = initialParties
         , utxoHash
         , contestationPeriod = initialContestationPeriod
         , headId = headIdToCurrencySymbol headId
         }
 
-  extractCommits d =
-    case fromScriptData d of
-      Nothing -> error "SNAFU"
+  extractCommits txOut =
+    case txOutScriptData (toTxContext txOut) >>= fromScriptData of
+      Nothing -> error "Expected inline commit datum" -- TODO: proper error handling
```

Comment:
shall we resolve this TODO before merging?

Reply:
No, see my PR description for why.